lib/ansible/modules/cloud/amazon/ec2_vpc_vpn.py

```diff
@@ -417,8 +436,22 @@ def find_connection_response(connections=None):
             return connections['VpnConnections'][0]
 
 
-def create_connection(connection, customer_gateway_id, static_only, vpn_gateway_id, connection_type):
+def create_connection(connection, customer_gateway_id, static_only, vpn_gateway_id, connection_type, tunnel_options=None):
     """ Creates a VPN connection """
+
+    options = {'StaticRoutesOnly': static_only}
+
+    if isinstance(tunnel_options, list) and 0 < len(tunnel_options) <= 2:
```

Comment:
tunnel_options will never not be a list here since it's defaulting to an empty list and has type='list' in the arg spec so you could simplify this a bit to:`if tunnel_options and len(tunnel_options) <=2:`

However, would it be better not to silently ignore if you list more than 2? The error that would be returned by boto is probably enough in terms of validation for this option (though I haven't tested it, and that should be checked). If it isn't caught and handled properly, `botocore.exceptions.BotoCoreError` handling may need to be added to the call in addition to the botocore.exceptions.ClientError (the difference being that BotoCoreError has no `.response` attribute).

Reply:
Regarding your first point on 'simplification', I agree and will massage the code accordingly.

Regarding the list size, my original thinking was that if someone is doing something funky (e.g: sending more dicts than Boto3/AWS are capable of handling), we should just ignore it.  I could see throwing a harmless warning, advising the executor that only up to 2 pairs of options are supported.  I am, however, open to throwing a fatal exception as you indicated.

Reply:
I don't have a strong preference. LGTM!